src/test/java/org/jabref/logic/importer/fetcher/CompositeIdFetcherTest.java

```diff
@@ -121,4 +121,16 @@ void performSearchByIdReturnsEmptyForInvalidId(String groundInvalidArXivId) thro
     void performSearchByIdReturnsCorrectEntryForIdentifier(String name, BibEntry bibEntry, String identifier) throws FetcherException {
         assertEquals(Optional.of(bibEntry), compositeIdFetcher.performSearchById(identifier));
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = "https://www.scitepress.org/Link.aspx?doi=10.5220/0010404301780189")
```

Comment:
The "secret" sauce for parameterized tests is to pass both the expected result and the input.

Try

```java
@CsvSource({"10.5220/0010404301780189,"https://www.scitepress.org/Link.aspx?doi=10.5220/0010404301780189", 
"10.5220/0010404301780189,10.5220/0010404301780189"})
 void filterForDOITagReturnsDOI(String expected, String DOIString) throws FetcherException {
```

Oh, the test seems to be go after https://github.com/koppor/jabref/blob/main/src/test/java/org/jabref/model/entry/identifier/DOITest.java#L203-L203

Reply:
Ok, I've read up a bit on CsvSource-parameterized tests, but I don't understand why this doesn't work:

```@ParameterizedTest
    @CsvSource({"https://www.scitepress.org/Link.aspx?doi=10.5220/0010404301780189, 10.5220/0010404301780189",
                "10.5220/0010404301780189, 10.5220/0010404301780189"})
    void performSearchByIdRemovesDOITag(String input, String expectedResult) throws FetcherException {
        assertEquals(expectedResult, compositeIdFetcher.performSearchById(input));
    }

Reply:
Ah, I didn't read closely enough....

Reply:
Nope. It seems to me this should work, though of course it doesn't:

```@ParameterizedTest
    @CsvSource({"10.5220/0010404301780189, https://www.scitepress.org/Link.aspx?doi=10.5220/0010404301780189",
                "10.5220/0010404301780189, 10.5220/0010404301780189"})
    void performSearchByIdRemovesDOITag(String expectedResult, String input) throws FetcherException {
        assertEquals(expectedResult, compositeIdFetcher.performSearchById(input));
    }

Reply:
"it doesn't" is too unspecific for me.

I was about to write follwing

- If you set a breakpoint at `assertEquals` and press the debug button. Which values do `input` and `expectedResult` take?
- Then highlight `compositeIdFetcher.performSearchById(input)`
- Then press <kbd>Alt</kbd>+<kbd>F8</kbd> to evaluate 

While writing, it is clear: ´compositeIdFetcher.performSearchById(input)` performs a search on the string. You just want to test the parsing of the ID there.

I am so sorry, I was not clear what I meant with "Oh, the test seems to be go"

I meant: Forget my eduacation about CSV source. Just add two lines after https://github.com/koppor/jabref/blob/main/src/test/java/org/jabref/model/entry/identifier/DOITest.java#L203-L203

I write them for you:

```
 Arguments.of("10.5220/0010404301780189", DOI.findInText("https://www.scitepress.org/Link.aspx?doi=10.5220/0010404301780189").get().getDOI()),
 Arguments.of("10.5220/0010404301780189", DOI.findInText("10.5220/0010404301780189").get().getDOI()),
 ```

---

Education:

I know, these tests are not an example for good tests. They should be rewritten so that only pairs of expected/input-doi-string are passed. For that, however, the tests need to split up. Needs concentration.

Reply:
> If you set a breakpoint at assertEquals and press the debug button. Which values do input and expectedResult take?
Then highlight compositeIdFetcher.performSearchById(input)
Then press Alt+F8 to evaluate

Well, that was certainly idiotic of me since DOI.findInText returns an Optional and therefore the assertEquals comparing strings will fail in any case. Thanks for the patient explanation!